File: codetree/프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    map1 = []
    map2 = []
    
    new_arr = [[0]*n for _ in range(n)]
    
    for i in range(n):
        cur_line1 = format(arr1[i], f'0{n}b')
        cur_line2 = format(arr2[i], f'0{n}b')
        map1.append(list(map(int, cur_line1)))
        map2.append(list(map(int, cur_line2)))
        
    for i in range(n):
        for j in range(n):
            new_arr[i][j] = map1[i][j]|map2[i][j]
    
    for i in range(n):
        tmp = ""
        for j in range(n):
            if new_arr[i][j]==1:
                tmp+='#'
            else:
                tmp+=' '
        answer.append(tmp)
    
    return answer

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.*;

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        ArrayList<String> answer = new ArrayList<>();
        
        int[][] new_arr1 = new int[n][n];
        int[][] new_arr2 = new int[n][n];
        int[][] answer_arr = new int[n][n];
        
        for (int i=0; i<n; i++) {
            String binary = Integer.toBinaryString(arr1[i]);
            while (binary.length() < n) {
                binary = "0" + binary;
            }
            // System.out.println(binary);
            
            for (int j=0; j<n; j++) {
                new_arr1[i][j] = binary.charAt(j)-'0';
            }
        }
        
        for (int i=0; i<n; i++) {
            String binary = Integer.toBinaryString(arr2[i]);
            while (binary.length() < n) {
                binary = "0" + binary;
            }
            // System.out.println(binary);
            
            for (int j=0; j<n; j++) {
                new_arr2[i][j] = binary.charAt(j)-'0';
            }
        }
        
        for (int i=0; i<n; i++) {
            for (int j=0; j<n; j++) {
                if (new_arr1[i][j]==1 || new_arr2[i][j]==1) {
                    answer_arr[i][j]=1;
                }
                else {
                    answer_arr[i][j]=0;
                }
            }
            //System.out.println();
        }
        
        for (int i=0; i<n; i++) {
            StringBuilder tmp = new StringBuilder();
            for (int j=0; j<n; j++) {
                if (answer_arr[i][j]==1) tmp.append("#");
                else tmp.append(" ");
            }
            answer.add(tmp.toString());
        }
        
        return answer.toArray(new String[0]);
    }
}